In [9]:
import os
import sys
import pprint
import pandas as pd
import numpy as np

base_dir = '/media/tord/T7/Thesis_ssd/MasterThesis3'
os.chdir(base_dir)

from GlobalUtils import GlobalUtils

from Classes.DataProcessing.LoadData import LoadData
from Classes.DataProcessing.DataHandler import DataHandler
from Classes.DataProcessing.DataGenerator import DataGenerator
from Classes.Modeling.GridSearchResultProcessor import GridSearchResultProcessor
from Classes.Modeling.CustomCallback import CustomCallback
from Classes.Scaling.ScalerFitter import ScalerFitter
from Classes.Scaling.MinMaxScalerFitter import MinMaxScalerFitter
from Classes.Scaling.StandardScalerFitter import StandardScalerFitter
from Classes.Modeling.ResultFitter import ResultFitter
import json

utils = GlobalUtils()

In [13]:
def clear_nans(result_file_name, num_classes):
    resultProcessor = GridSearchResultProcessor()
    df = resultProcessor.get_results_df_by_name(result_file_name, num_classes)
    df_copy = df.copy()
    no_nans = df_copy.dropna()
    clear_results_df(result_file_name, num_classes)
    save_results_df(no_nans, result_file_name, num_classes)

def save_results_df(results_df, file_name, num_classes):
    results_df.to_csv(f"{get_results_file_path(num_classes)}/{file_name}", mode = 'w', index=False)

def clear_results_df(file_name, num_classes):
    path = get_results_file_path(num_classes)
    file = f"{path}/{file_name}"
    if os.path.isfile(file):
        f = open(file, "w+")
        f.close()

def get_results_file_path(num_classes):
    utils = GlobalUtils()
    file_path = f'{utils.base_dir}/GridSearchResults/{num_classes}_classes'
    return file_path

clear_nans(result_file_name, 2)


In [15]:
def f1_score(precision, recall):
    f1 = 2*((precision*recall)/(precision + recall))
    return f1

def create_f1_list(precision_df, recall_df):
    f1 = []
    for i in range(len(precision_df)):
        f1.append(f1_score(precision_df.loc[i], recall_df.loc[i]))
    return f1

#result_file_name = 'results_InceptionTime_noiseNotNoise_timeAug_sscale_noiseAug_earlyS.csv'
result_file_name = 'results_InceptionTime_earthExplo_timeAug_sscale_noiseAug_earlyS.csv'
#result_file_name = 'results_InceptionTime_NARROW_noiseNotNoise_timeAug_sscale_noiseAug_earlyS.csv'
clear_nans(result_file_name, 2)
df = GridSearchResultProcessor().get_results_df_by_name(result_file_name, 2)

pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)
pd.set_option('display.width', None)
pd.set_option('display.max_colwidth', None)

df_f1 = df.copy()
df_f1.columns=df_f1.columns.str.strip()
all_train_precision = df_f1['train_precision']
all_train_recall = df_f1['train_recall']
all_val_precision = df_f1['val_precision']
all_val_recall = df_f1['val_recall']
f1_train = create_f1_list(all_train_precision, all_train_recall)
f1_val = create_f1_list(all_val_precision, all_val_recall)
df_f1['train_f1'] = f1_train
df_f1['val_f1'] = f1_val
df_f1

1.907417      0.504259       0.518473    0.974235  0.677889  0.676776  
114      0.690408      0.553607       0.576579    0.697806  0.633410  0.631426  
115      0.694560      0.607004       0.514893    0.025931  0.057439  0.049375  
116      0.687948      0.633218       0.516683    0.030419  0.067746  0.057455  
117      0.693522      0.489623       0.489665    1.000000  0.658722  0.657416  
118      0.584574      0.696984       0.708171    0.714594  0.824068  0.711368  
119      0.693136      0.000000       0.510417    0.000000  0.000000  0.000000  
120      0.620260      0.685591       0.678548    0.634309  0.733342  0.655683  
121      0.677205      0.551698       0.577555    0.731715  0.693550  0.645560  
122      1.717070      0.680464       0.687826    0.683178  0.978027  0.685494  
123      1.092375      0.547373       0.588837    0.932819  0.730501  0.721948  
124      0.663411      0.574360       0.588867    0.618850  0.659171  0.603486  
125     89.150085      0.000000       0.510417    0.000000  0.000000  0.000000  
126      0.610718      0.608809       0.653158    0.815658  0.948198  0.725419  
127      0.688782      0.528736       0.510824    0.007646  0.031782  0.015067  
128      0.670553      0.637255       0.619792    0.518617  0.643614  0.564708  
129     65.505730      0.000000       0.510417    0.000000  0.000000  0.000000  
130      0.682821      0.521133       0.548665    0.963265  0.717628  0.699120  
131   6136.697266      0.000000       0.509665    0.000000  0.000000  0.000000  
132      4.344726      0.522349       0.522868    0.297207  0.424322  0.378990  
133      1.518864      0.512789       0.533936    0.963098  0.705311  0.687002  
134      0.692617      0.000000       0.510417    0.000000  0.000000  0.000000  
135      0.659342      0.658948       0.629964    0.506150  0.624328  0.561311  
136    556.290344      0.000000       0.509928    0.000000  0.000196  0.000000  
137      0.697021      0.489583       0.489583    1.000000  0.659197  0.657343  
138      0.595907      0.723725       0.696126    0.613531  0.753035  0.652224  
139      0.693726      0.489901       0.490234    0.999834  0.660886  0.657893  
140      0.848720      0.576962       0.597087    0.663564  0.641146  0.628572  
141      0.725742      0.623873       0.584890    0.386334  0.558612  0.465316  
142     17.647678      0.000000       0.510417    0.000000  0.000000  0.000000  
143      0.686442      0.626087       0.512777    0.011968  0.055147  0.023390  
144      0.744100      0.000000       0.510417    0.000000  0.000000  0.000000  
145     40.064247      0.506750       0.522310    0.967970  0.684110  0.678504  
146      0.725455      0.000000       0.510417    0.000000  0.000000  0.000000  
147      0.684580      0.545455       0.509907    0.002957  0.006342  0.005879  
148    356.882721      0.000000       0.509665    0.000000  0.000000  0.000000  
149      0.596008      0.677783       0.697835    0.729721  0.797975  0.713422  
150      0.688638      0.585399       0.549613    0.279238  0.512706  0.370327  
151    541.184448      0.000000       0.510417    0.000000  0.000065  0.000000  
152      0.718351      0.554299       0.595622    0.888298  0.727532  0.713097  
153     23.839279      0.000000       0.510417    0.000000  0.000000  0.000000  
154      0.693298      0.489814       0.490072    0.999169  0.659057  0.657603  
155      0.692720      1.000000       0.509745    0.000164  0.006855  0.000328  
156      0.685038      0.428571       0.509928    0.002992  0.006416  0.005949  
157      0.615356      0.648923       0.667969    0.701130  0.888273  0.684148  
158      0.689107      0.448889       0.508545    0.016789  0.036988  0.032504  
159      1.377054      0.600000       0.510498    0.000499  0.000522  0.000996  
160      0.692719      0.000000       0.510417    0.000000  0.000000  0.000000  
161      0.682104      0.740260       0.512645    0.009363  0.017028  0.018390  
162      0.692790      0.494989       0.

In [16]:
sorted_f1_df = df_f1.sort_values(by=['val_f1', 'train_f1'], axis = 0, ascending=False)
sorted_f1_df

0.659108      0.567511       0.594564    0.722407  0.691781  0.652280  
138      0.595907      0.723725       0.696126    0.613531  0.753035  0.652224  
169      0.802952      0.703911       0.686050    0.620894  0.736865  0.651848  
309      5.379178      0.510987       0.528320    0.850399  0.741294  0.651740  
348      0.695312      0.772766       0.716173    0.596583  0.823554  0.650931  
36       0.595927      0.707540       0.687826    0.617686  0.671062  0.650872  
102      4.288969      0.636894       0.647868    0.653092  0.994051  0.650469  
311      0.672282      0.559365       0.585938    0.726729  0.725306  0.648783  
71       0.647176      0.648151       0.653239    0.638132  0.757387  0.645597  
121      0.677205      0.551698       0.577555    0.731715  0.693550  0.645560  
83       2.576699      0.551005       0.575277    0.715592  0.639089  0.637808  
269      0.652374      0.626088       0.635631    0.637812  0.678537  0.636720  
206      0.668416      0.553794       0.577799    0.708444  0.662315  0.636487  
114      0.690408      0.553607       0.576579    0.697806  0.633410  0.631426  
17       3.715873      0.631202       0.636598    0.622700  0.770361  0.629572  
214      0.644712      0.665158       0.656572    0.603318  0.684497  0.628820  
140      0.848720      0.576962       0.597087    0.663564  0.641146  0.628572  
26       2.221206      0.661196       0.654297    0.602726  0.907008  0.627454  
315      0.742686      0.596406       0.610840    0.634475  0.657136  0.622433  
182      1.184511      0.665237       0.654622    0.592919  0.762506  0.622245  
28       0.610758      0.740894       0.690186    0.564661  0.684885  0.621145  
197      0.692139      0.510619       0.525798    0.755319  0.616735  0.619998  
203      0.669759      0.557427       0.577555    0.665559  0.711590  0.618442  
108      0.673164      0.570032       0.588112    0.651117  0.651168  0.618013  
33       0.655151      0.588711       0.603271    0.629322  0.715524  0.616021  
179      0.838145      0.544894       0.564941    0.675864  0.669556  0.615445  
68       3.363179      0.658419       0.647874    0.585742  0.687348  0.615243  
41       1.601552      0.548220       0.567952    0.668052  0.620202  0.613949  
24       0.666789      0.557177       0.576091    0.653590  0.669685  0.612398  
189      0.656426      0.594008       0.605187    0.615473  0.661292  0.610287  
8        0.754115      0.566235       0.582764    0.631649  0.676914  0.606222  
9        0.680786      0.549088       0.567383    0.650765  0.604664  0.606220  
218      1.223141      0.600166       0.608317    0.599069  0.656371  0.603658  
124      0.663411      0.574360       0.588867    0.618850  0.659171  0.603486  
107      0.681620      0.544144       0.561768    0.646443  0.643000  0.601138  
271      0.690289      0.517713       0.532619    0.684133  0.598281  0.598942  
58       0.665242      0.582364       0.593424    0.599402  0.649175  0.596398  
245      5.802886      0.550577       0.566813    0.626995  0.594787  0.595387  
247      3.620691      0.725045       0.671875    0.531250  0.735696  0.593344  
106      1.065478      0.540098       0.554607    0.617280  0.588914  0.584268  
76       0.638702      0.659661       0.637777    0.537400  0.647343  0.583302  
255      0.690918      0.509402       0.522298    0.657414  0.621740  0.582118  
192      0.658671      0.601860       0.601318    0.548537  0.714421  0.573716  
215      0.659729      0.783219       0.684570    0.491855  0.708147  0.572428  
281      0.675374      0.568519       0.577230    0.566157  0.683284  0.571640  
16       0.736633      0.687977       0.646810    0.509807  0.794654  0.570194  
94       0.674432      0.562337       0.571360    0.567510  0.566945  0.569428  
128      0.670553      0.637255       0.619792    0.518617  0.643614  0.564708  
90      10.466121      0.507741       0.519531    0.610539  0.570186  0.561370  
135      0.659342      0.658948       0.

In [5]:
sorted_f1_df = df_f1.sort_values(by=['val_precision', 'train_precision'], axis = 0, ascending=False)
sorted_f1_df

2.799467      0.548781       0.511068    0.007480  0.024575  0.014744  
283      0.684387      0.531532       0.510986    0.009807  0.028853  0.019245  
402      0.692902      0.504414       0.510986    0.066489  0.116459  0.117668  
79      13.530292      0.585366       0.510986    0.003989  0.008726  0.007917  
153      0.688782      0.528736       0.510824    0.007646  0.031782  0.015067  
16       0.903282      0.666667       0.510742    0.001330  0.006353  0.002653  
134      0.708781      0.500084       0.510579    0.991855  0.714825  0.674134  
325      8.032729      1.000000       0.510579    0.000332  0.000391  0.000664  
196    170.556091      0.503200       0.510551    0.142083  0.225413  0.222301  
347      0.691854      0.538462       0.510498    0.001164  0.003638  0.002322  
189      1.377054      0.600000       0.510498    0.000499  0.000522  0.000996  
272      4.157801      0.600000       0.510498    0.000499  0.000391  0.000996  
415      0.693359      1.000000       0.510498    0.000166  0.000065  0.000332  
320      0.705412      0.000000       0.510417    0.000000  0.000000  0.000000  
306      5.875718      0.000000       0.510417    0.000000  0.000065  0.000000  
123      0.692790      0.000000       0.510417    0.000000  0.000000  0.000000  
135      8.160054      0.000000       0.510417    0.000000  0.000000  0.000000  
31       3.528035      0.000000       0.510417    0.000000  0.000000  0.000000  
246      3.663452      0.000000       0.510417    0.000000  0.000000  0.000000  
216      6.067274      0.000000       0.510417    0.000000  0.000000  0.000000  
95      15.819236      0.000000       0.510417    0.000000  0.000000  0.000000  
293      2.122535      0.000000       0.510417    0.000000  0.000000  0.000000  
22       1.241366      0.000000       0.510417    0.000000  0.000000  0.000000  
88       1.822548      0.000000       0.510417    0.000000  0.000000  0.000000  
91       0.692749      0.000000       0.510417    0.000000  0.000000  0.000000  
318     38.965084      0.000000       0.510417    0.000000  0.000000  0.000000  
271   1836.532227      0.000000       0.510417    0.000000  0.000000  0.000000  
326     12.688478      0.000000       0.510417    0.000000  0.000000  0.000000  
85       0.722922      0.000000       0.510417    0.000000  0.000196  0.000000  
286      5.690711      0.000000       0.510417    0.000000  0.000000  0.000000  
126      1.064214      0.000000       0.510417    0.000000  0.000000  0.000000  
269      0.692647      0.000000       0.510417    0.000000  0.000000  0.000000  
289     85.831596      0.000000       0.510417    0.000000  0.000000  0.000000  
247      0.691508      0.000000       0.510417    0.000000  0.000000  0.000000  
155     65.505730      0.000000       0.510417    0.000000  0.000000  0.000000  
168     17.647678      0.000000       0.510417    0.000000  0.000000  0.000000  
198   3406.155029      0.000000       0.510417    0.000000  0.000000  0.000000  
15      77.859322      0.000000       0.510417    0.000000  0.000000  0.000000  
182     23.839279      0.000000       0.510417    0.000000  0.000000  0.000000  
190      0.692719      0.000000       0.510417    0.000000  0.000000  0.000000  
259   1460.758179      0.000000       0.510417    0.000000  0.000000  0.000000  
278      4.110057      0.000000       0.510417    0.000000  0.000000  0.000000  
307     29.640028      0.500000       0.510417    0.013464  0.023668  0.026236  
71       0.692576      0.000000       0.510417    0.000000  0.000000  0.000000  
151     89.150085      0.000000       0.510417    0.000000  0.000000  0.000000  
294  19042.556641      0.000000       0.510417    0.000000  0.000000  0.000000  
118      0.866035      0.000000       0.510417    0.000000  0.000065  0.000000  
160      0.692617      0.000000       0.510417    0.000000  0.000000  0.000000  
291      1.973031      0.000000       0.510417    0.000000  0.000000  0.000000  
9        0.693013      0.000000       0.

In [6]:
sorted_f1_df = df_f1.sort_values(by=['val_loss', 'train_loss'], axis = 0, ascending=True)
sorted_f1_df

0.722922      0.000000       0.510417    0.000000  0.000196  0.000000  
172      0.725455      0.000000       0.510417    0.000000  0.000000  0.000000  
167      0.725742      0.623873       0.584890    0.386334  0.558612  0.465316  
372      0.728764      0.489583       0.489583    1.000000  0.658765  0.657343  
20       0.736633      0.687977       0.646810    0.509807  0.794654  0.570194  
368      0.742686      0.596406       0.610840    0.634475  0.657136  0.622433  
261      0.743614      0.000000       0.509665    0.000000  0.000000  0.000000  
170      0.744100      0.000000       0.510417    0.000000  0.000000  0.000000  
10       0.754115      0.566235       0.582764    0.631649  0.676914  0.606222  
111      0.755208      0.000000       0.509665    0.000000  0.000000  0.000000  
23       0.755321      0.440407       0.503061    0.049770  0.127210  0.090579  
218      0.773977      0.532658       0.568115    0.961104  0.719376  0.714113  
121      0.777162      0.000000       0.510417    0.000000  0.000000  0.000000  
396      0.800784      0.492624       0.495931    0.988032  0.664021  0.660388  
202      0.802952      0.703911       0.686050    0.620894  0.736865  0.651848  
212      0.838145      0.544894       0.564941    0.675864  0.669556  0.615445  
299      0.844766      0.486500       0.501383    0.332447  0.424367  0.399802  
166      0.848720      0.576962       0.597087    0.663564  0.641146  0.628572  
197      0.857544      0.549261       0.554204    0.506406  0.643259  0.529228  
75       0.860223      0.503721       0.516672    0.967148  0.689206  0.673530  
8        0.861358      0.000000       0.509665    0.000000  0.000000  0.000000  
118      0.866035      0.000000       0.510417    0.000000  0.000065  0.000000  
249      0.880798      0.491689       0.508870    0.093418  0.153481  0.157856  
16       0.903282      0.666667       0.510742    0.001330  0.006353  0.002653  
209      0.911945      0.489583       0.489583    1.000000  0.658218  0.657343  
219      0.920779      0.675526       0.680257    0.667553  0.729457  0.673845  
391      0.929764      0.489620       0.489665    0.999668  0.659060  0.657344  
45       0.935396      0.333333       0.509343    0.000657  0.001495  0.001312  
132      0.940804      0.000000       0.509504    0.000000  0.000195  0.000000  
41       0.964536      0.566941       0.607178    0.836935  0.792089  0.703779  
232      0.966666      0.636731       0.675700    0.786070  0.883654  0.726718  
420      0.989058      0.562825       0.606852    0.882314  0.764236  0.719106  
358      0.997266      0.640268       0.544434    0.158577  0.279525  0.245614  
386      1.015954      0.400000       0.510173    0.000997  0.003511  0.001991  
105      1.053267      0.000000       0.509665    0.000000  0.000000  0.000000  
126      1.064214      0.000000       0.510417    0.000000  0.000000  0.000000  
129      1.065478      0.540098       0.554607    0.617280  0.588914  0.584268  
119      1.083119      0.490617       0.491781    0.995180  0.663578  0.658269  
149      1.092375      0.547373       0.588837    0.932819  0.730501  0.721948  
280      1.128404      0.000000       0.510417    0.000000  0.000000  0.000000  
348      1.133954      0.529775       0.562419    0.957129  0.755991  0.708510  
65       1.145252      0.613700       0.636312    0.693983  0.757595  0.663897  
389      1.163020      0.000000       0.509665    0.000000  0.000000  0.000000  
215      1.184511      0.665237       0.654622    0.592919  0.762506  0.622245  
100      1.186945      0.570423       0.513672    0.026928  0.055099  0.051174  
405      1.194230      0.606723       0.646322    0.789062  0.825143  0.710595  
256      1.223141      0.600166       0.608317    0.599069  0.656371  0.603658  
22       1.241366      0.000000       0.510417    0.000000  0.000000  0.000000  
227      1.292256      0.000000       0.510335    0.000000  0.000326  0.000000  
414      1.303931      0.567291       0.

In [7]:
sorted_f1_df = df_f1.sort_values(by=['val_accuracy', 'train_accuracy'], axis = 0, ascending=False)
sorted_f1_df

1.518864      0.512789       0.533936    0.963098  0.705311  0.687002  
362      5.379178      0.510987       0.528320    0.850399  0.741294  0.651740  
419     49.726620      0.510826       0.530436    0.964761  0.719050  0.684517  
231      0.692139      0.510619       0.525798    0.755319  0.616735  0.619998  
13      10.835087      0.509794       0.516764    0.337434  0.430154  0.408275  
298      0.690918      0.509402       0.522298    0.657414  0.621740  0.582118  
87       1.487955      0.508365       0.525635    0.944481  0.680405  0.675392  
109     10.466121      0.507741       0.519531    0.610539  0.570186  0.561370  
322      0.698524      0.507209       0.523196    0.970762  0.685796  0.679937  
171     40.064247      0.506750       0.522310    0.967970  0.684110  0.678504  
228      0.691935      0.505270       0.520264    0.964262  0.680197  0.675866  
402      0.692902      0.504414       0.510986    0.066489  0.116459  0.117668  
136      1.907417      0.504259       0.518473    0.974235  0.677889  0.676776  
330      7.800753      0.503855       0.517253    0.912400  0.663102  0.660218  
75       0.860223      0.503721       0.516672    0.967148  0.689206  0.673530  
196    170.556091      0.503200       0.510551    0.142083  0.225413  0.222301  
134      0.708781      0.500084       0.510579    0.991855  0.714825  0.674134  
307     29.640028      0.500000       0.510417    0.013464  0.023668  0.026236  
377      0.689357      0.500000       0.509665    0.001314  0.001690  0.002621  
206      2.640077      0.498074       0.505960    0.976840  0.670189  0.666633  
243      5.394547      0.496482       0.503581    0.985206  0.669684  0.666490  
296      0.700195      0.496250       0.503174    0.978890  0.667576  0.664684  
192      0.692790      0.494989       0.500651    0.985206  0.666496  0.663919  
384     78.757370      0.494513       0.499674    0.988697  0.673942  0.663849  
285      2.047912      0.493321       0.497314    0.988364  0.673668  0.661688  
315      0.698036      0.492780       0.496257    0.987035  0.661976  0.660453  
264      6.034113      0.492761       0.496175    0.990027  0.663930  0.661050  
396      0.800784      0.492624       0.495931    0.988032  0.664021  0.660388  
213      4.699515      0.492261       0.505046    0.348903  0.423749  0.412699  
385      0.704346      0.492247       0.504313    0.395778  0.444711  0.443502  
249      0.880798      0.491689       0.508870    0.093418  0.153481  0.157856  
3       12.157683      0.490844       0.492188    0.998005  0.659642  0.659251  
379  31691.082031      0.490749       0.491221    0.997700  0.658436  0.658317  
119      1.083119      0.490617       0.491781    0.995180  0.663578  0.658269  
341     90.944687      0.490487       0.490657    0.999343  0.658251  0.658167  
277    110.536598      0.490335       0.490335    1.000000  0.658249  0.658020  
165      0.693726      0.489901       0.490234    0.999834  0.660886  0.657893  
183      0.693298      0.489814       0.490072    0.999169  0.659057  0.657603  
140      0.693522      0.489623       0.489665    1.000000  0.658722  0.657416  
391      0.929764      0.489620       0.489665    0.999668  0.659060  0.657344  
378      0.708822      0.489583       0.489583    1.000000  0.659139  0.657343  
372      0.728764      0.489583       0.489583    1.000000  0.658765  0.657343  
410      0.694356      0.489583       0.489583    1.000000  0.658549  0.657343  
163      0.697021      0.489583       0.489583    1.000000  0.659197  0.657343  
275      0.693359      0.489583       0.489583    1.000000  0.659082  0.657343  
284      0.693237      0.489583       0.489583    1.000000  0.658923  0.657343  
210  40245.402344      0.489583       0.489583    1.000000  0.658866  0.657343  
92       0.693095      0.489583       0.489583    1.000000  0.658751  0.657343  
220    140.757690      0.489583       0.489583    1.000000  0.658693  0.657343  
223      5.112727      0.489583       0.

## Findings:
 - New optimal dropout_rate: 0.2 (train_acc: 98%, val_acc: 90%) Old 0.3 (train_acc: 96.8%, val_acc: 90.9%)
 - RELU and Sigmoid activation functions do not work with this model.
 - L1 regularization is counter productive (Less is more) so let l1 = 0.00
 - 32 start neurons appear to be optimal

In [161]:
df.iloc[40:]

batch_size  epochs  learning_rate  num_layers optimizer activation  \
40  64          33      0.20000        2           sgd       tanh        
41  64          33      0.10000        2           sgd       tanh        
42  64          33      0.00100        2           sgd       tanh        
43  64          33      0.00010        2           sgd       tanh        
44  64          33      0.00001        2           sgd       tanh        
45  64          33      0.01000        2           sgd       tanh        
46  64          33      0.01000        2           sgd       tanh        
47  64          33      0.01000        2           sgd       tanh        
48  64          33      0.01000        2           sgd       tanh        
49  64          33      0.01000        2           sgd       tanh        
50  64          33      0.01000        2           sgd       tanh        
51  64          33      0.01000        2           sgd       tanh        
52  64          33      0.01000        2           sgd       tanh        
53  64          33      0.01000        2           sgd       tanh        
54  32          33      0.01000        3           sgd       tanh        
55  32          33      0.01000        4           sgd       tanh        
56  32          33      0.01000        5           sgd       tanh        
57  32          33      0.01000        6           sgd       tanh        
58  24          33      0.01000        2           sgd       tanh        
59  28          33      0.01000        2           sgd       tanh        
60  34          33      0.01000        2           sgd       tanh        
61  36          33      0.01000        2           sgd       tanh        
62  32          33      0.10000        2           sgd       tanh        
63  32          33      0.05000        2           sgd       tanh        
64  32          33      0.02500        2           sgd       tanh        
65  32          33      0.00500        2           sgd       tanh        

         decay_sequence  dropout_rate  filters  kernel_size    l1_r   l2_r  \
40  [1, 2]               0.3           17       5            0.0001  0.001   
41  [1, 2]               0.3           17       5            0.0001  0.001   
42  [1, 2]               0.3           17       5            0.0001  0.001   
43  [1, 2]               0.3           17       5            0.0001  0.001   
44  [1, 2]               0.3           17       5            0.0001  0.001   
45  [1, 2]               0.3           17       5            0.0001  0.001   
46  [1, 2]               0.3           17       5            0.0001  0.001   
47  [1, 2]               0.3           17       5            0.0001  0.001   
48  [1, 2]               0.3           17       5            0.0001  0.001   
49  [1, 2]               0.3           17       5            0.0001  0.001   
50  [1, 2]               0.3           17       5            0.0001  0.001   
51  [1, 4]               0.3           17       5            0.0001  0.001   
52  [1, 0.5]             0.3           17       5            0.0001  0.001   
53  [1, 1]               0.3           17       5            0.0001  0.001   
54  [1, 2, 4]            0.3           17       5            0.0001  0.001   
55  [1, 2, 4, 6]         0.3           17       5            0.0001  0.001   
56  [1, 2, 4, 6, 8]      0.3           17       5            0.0001  0.001   
57  [1, 2, 4, 6, 8, 10]  0.3           17       5            0.0001  0.001   
58  [1, 2]               0.3           17       5            0.0001  0.001   
59  [1, 2]               0.3           17       5            0.0001  0.001   
60  [1, 2]               0.3           17       5            0.0001  0.001   
61  [1, 2]               0.3           17       5            0.0001  0.001   
62  [1, 2]               0.3           17       5            0.0001  0.001   
63  [1, 2]               0.3           17       5            0.0001  0.001   
64  [1, 2]               0.3           17       5      

In [162]:
def f1_score(precision, recall):
    f1 = 2*((precision*recall)/(precision + recall))
    return f1

def create_f1_list(precision_df, recall_df):
    f1 = []
    for i in range(len(precision_df)):
        f1.append(f1_score(precision_df.loc[i], recall_df.loc[i]))
    return f1

batch_size  epochs  learning_rate  num_layers optimizer activation  \
0   64          33      0.01000        2           sgd       tanh        
1   64          33      0.01000        2           sgd       tanh        
2   64          33      0.01000        2           sgd       tanh        
3   64          33      0.01000        2           sgd       tanh        
4   64          33      0.01000        2           sgd       tanh        
5   64          33      0.01000        2           sgd       tanh        
6   64          33      0.01000        2           sgd       tanh        
7   64          33      0.01000        2           sgd       tanh        
8   64          33      0.01000        2           adam      tanh        
9   64          33      0.01000        2           rmsprop   tanh        
10  64          33      0.01000        2           sgd       tanh        
11  64          33      0.01000        2           sgd       tanh        
12  64          33      0.01000        2           sgd       tanh        
13  64          33      0.01000        2           sgd       tanh        
14  64          33      0.01000        2           sgd       tanh        
15  64          33      0.01000        2           sgd       tanh        
16  64          33      0.01000        2           sgd       tanh        
17  64          33      0.01000        2           sgd       tanh        
18  64          29      0.01000        2           sgd       tanh        
19  64          37      0.01000        2           sgd       tanh        
20  64          33      0.01000        2           sgd       tanh        
21  64          33      0.01000        2           sgd       tanh        
22  64          33      0.01000        2           sgd       tanh        
23  64          33      0.01000        2           sgd       tanh        
24  64          33      0.01000        2           sgd       tanh        
25  64          33      0.01000        2           sgd       tanh        
26  64          33      0.01000        2           sgd       relu        
27  64          33      0.01000        2           sgd       sigmoid     
28  64          33      0.01000        2           sgd       softmax     
29  64          33      0.01000        2           sgd       tanh        
30  64          33      0.01000        2           sgd       tanh        
31  64          33      0.01000        1           sgd       tanh        
32  64          33      0.01000        3           sgd       tanh        
33  64          33      0.01000        4           sgd       tanh        
34  64          33      0.01000        5           sgd       tanh        
35  64          33      0.01000        6           sgd       tanh        
36  32          33      0.01000        2           sgd       tanh        
37  128         33      0.01000        2           sgd       tanh        
38  256         33      0.01000        2           sgd       tanh        
39  512         33      0.01000        2           sgd       tanh        
40  64          33      0.20000        2           sgd       tanh        
41  64          33      0.10000        2           sgd       tanh        
42  64          33      0.00100        2           sgd       tanh        
43  64          33      0.00010        2           sgd       tanh        
44  64          33      0.00001        2           sgd       tanh        
45  64          33      0.01000        2           sgd       tanh        
46  64          33      0.01000        2           sgd       tanh        
47  64          33      0.01000        2           sgd       tanh        
48  64          33      0.01000        2           sgd       tanh        
49  64          33      0.01000        2           sgd       tanh        
50  64          33      0.01000        2           sgd       tanh        
51  64          33      0.01000        2           sgd       tanh        
52  64          33      0.01000        2           sgd       tanh        
53  64  

In [170]:
sorted_f1_df

batch_size  epochs  learning_rate  num_layers optimizer activation  \
19  64          37      0.01000        2           sgd       tanh        
10  64          33      0.01000        2           sgd       tanh        
25  64          33      0.01000        2           sgd       tanh        
11  64          33      0.01000        2           sgd       tanh        
13  64          33      0.01000        2           sgd       tanh        
47  64          33      0.01000        2           sgd       tanh        
36  32          33      0.01000        2           sgd       tanh        
51  64          33      0.01000        2           sgd       tanh        
14  64          33      0.01000        2           sgd       tanh        
29  64          33      0.01000        2           sgd       tanh        
49  64          33      0.01000        2           sgd       tanh        
48  64          33      0.01000        2           sgd       tanh        
37  128         33      0.01000        2           sgd       tanh        
3   64          33      0.01000        2           sgd       tanh        
53  64          33      0.01000        2           sgd       tanh        
12  64          33      0.01000        2           sgd       tanh        
32  64          33      0.01000        3           sgd       tanh        
40  64          33      0.20000        2           sgd       tanh        
4   64          33      0.01000        2           sgd       tanh        
45  64          33      0.01000        2           sgd       tanh        
18  64          29      0.01000        2           sgd       tanh        
46  64          33      0.01000        2           sgd       tanh        
50  64          33      0.01000        2           sgd       tanh        
5   64          33      0.01000        2           sgd       tanh        
52  64          33      0.01000        2           sgd       tanh        
2   64          33      0.01000        2           sgd       tanh        
34  64          33      0.01000        5           sgd       tanh        
0   64          33      0.01000        2           sgd       tanh        
6   64          33      0.01000        2           sgd       tanh        
62  32          33      0.10000        2           sgd       tanh        
7   64          33      0.01000        2           sgd       tanh        
38  256         33      0.01000        2           sgd       tanh        
41  64          33      0.10000        2           sgd       tanh        
15  64          33      0.01000        2           sgd       tanh        
35  64          33      0.01000        6           sgd       tanh        
59  28          33      0.01000        2           sgd       tanh        
24  64          33      0.01000        2           sgd       tanh        
1   64          33      0.01000        2           sgd       tanh        
16  64          33      0.01000        2           sgd       tanh        
27  64          33      0.01000        2           sgd       sigmoid     
23  64          33      0.01000        2           sgd       tanh        
28  64          33      0.01000        2           sgd       softmax     
22  64          33      0.01000        2           sgd       tanh        
21  64          33      0.01000        2           sgd       tanh        
20  64          33      0.01000        2           sgd       tanh        
26  64          33      0.01000        2           sgd       relu        
61  36          33      0.01000        2           sgd       tanh        
17  64          33      0.01000        2           sgd       tanh        
60  34          33      0.01000        2           sgd       tanh        
39  512         33      0.01000        2           sgd       tanh        
42  64          33      0.00100        2           sgd       tanh        
9   64          33      0.01000        2           rmsprop   tanh        
33  64          33      0.01000        4           sgd       tanh        
54  32  

In [155]:
pd.to_numeric(df_f1['val_f1'], errors='coerce')

0     0.821132
1     0.804132
2     0.824117
3     0.832628
4     0.830515
5     0.828167
6     0.820769
7     0.818637
8     0.766957
9     0.785517
10    0.850172
11    0.843275
12    0.831845
13    0.840825
14    0.836419
15    0.809115
16    0.800293
17    0.790123
18    0.828972
19    0.856512
20    0.796238
21    0.796238
22    0.796238
23    0.796238
24    0.805288
25    0.845431
26    0.796238
27    0.796238
28    0.796238
29    0.836253
30   NaN      
31    0.761004
32    0.831517
33    0.784201
34    0.821750
35    0.806797
36    0.837878
37    0.833272
38    0.818396
39    0.789155
40    0.831131
41    0.815500
42    0.789035
43    0.583601
44    0.572993
45    0.829969
46    0.828561
47    0.838636
48    0.833555
49    0.835149
50    0.828306
51    0.836628
52    0.825959
53    0.832174
54    0.780612
55    0.775374
56    0.756599
57    0.712715
58    0.754318
59    0.806134
60    0.789657
61    0.795973
62    0.819888
63    0.761058
64   NaN      
Name: val_f1, dtype: floa

In [152]:
df_f1['val_f1'].isnull().values.any()

True

In [156]:
df_f1.columns

Index(['batch_size', 'epochs', 'learning_rate', 'num_layers', 'optimizer',
       'activation', 'decay_sequence', 'dropout_rate', 'filters',
       'kernel_size', 'l1_r', 'l2_r', 'output_layer_activation', 'padding',
       'start_neurons', 'train_loss', 'train_precision', 'train_accuracy',
       'train_recall', 'val_loss', 'val_precision', 'val_accuracy',
       'val_recall', 'train_f1', 'val_f1'],
      dtype='object')